src/Microsoft.DotNet.Arcade.Sdk/src/PipelinesLogger.cs

```diff
@@ -0,0 +1,422 @@
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Arcade.Sdk.src
+{
+    /// <summary>
+    /// Logger for converting MSBuild error messages to the Azure Pipelines Tasks format
+    /// 
+    /// https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md
+    /// </summary>
+    public sealed class PipelinesLogger : ILogger
+    {
+        private readonly MessageBuilder _builder = new MessageBuilder();
+        private readonly Dictionary<BuildEventContext, Guid> _buildEventContextMap = new Dictionary<BuildEventContext, Guid>(BuildEventContextComparer.Instance);
+        private readonly Dictionary<Guid, ProjectInfo> _projectInfoMap = new Dictionary<Guid, ProjectInfo>();
+        private readonly HashSet<Guid> _detailedLoggedSet = new HashSet<Guid>();
+        private HashSet<string> _ignoredTargets;
+        private string _solutionDirectory;
+
+        public LoggerVerbosity Verbosity { get; set; }
+        public string Parameters { get; set; }
+
+        public void Initialize(IEventSource eventSource)
+        {
+            var parameters = LoggerParameters.Parse(this.Parameters);
+
+            _solutionDirectory = parameters["SolutionDir"];
+
+            var verbosityString = parameters["Verbosity"];
+            Verbosity = !string.IsNullOrEmpty(verbosityString) && Enum.TryParse(verbosityString, out LoggerVerbosity verbosity)
+                ? verbosity
+                : LoggerVerbosity.Normal;
+
+            var ignoredTargets = new string[]
+            {
+                "GetCopyToOutputDirectoryItems",
+                "GetNativeManifest",
+                "GetTargetPath",
```

Comment:
```suggestion
                "GetTargetPath",
                "GetTargetFrameworks",
```

(new for multitargeting)